### PR TITLE
Add grid websocket approach

### PR DESCRIPTION
**Pull request by IonesioJunior**
Server app and client working properly using HTTP protocol + Socket IO communication pattern 
- Async websocket grid app
- Sync websocket grid client

**PS:** I noticed that @mccorby  had already implemented a client version using the socket io approach. I made only a few adaptations to fit the pattern that was adopted in this repository._(Thanks @mccorby :smiley: )_ . 

**TODO:**
- Add Heroku  deployment settings.
- Improve a security level using the HTTPS protocol _(It is apparently already enabled, but the creator of the socket io library does not provide in the documentation a way to test with self-signed certificates for security reasons)_. But it should work with certificates signed by certifying entities
**Issue link:** https://github.com/miguelgrinberg/python-socketio/issues/283

### CODE REVIEW

**Comment by mccorby** [comment]
Good one!
Be aware that this implementation blocks the server socket. To allow the same socket to be used by more than one client we would need to modify the way the flags (they should be `threading.Event()`) do the blocking.
My idea is to create an object per client so that the server would block depending on the client identifier (so every operation coming through `socketio` should include the client's id)

**Comment by mari-linhares** [comment]
Related to #200

**Comment by IonesioJunior** [comment]
**Branch status Update**:
- Deploy on Heroku : **DONE**
- Perform HTTPS connections: **DONE** _(The client is already compatible with the protocol.)_
- Add eventlet worker proxy to manage balance and control concurrency

Problems to solve:
I send and retrieve the tensors correctly. However, when trying to perform arithmetic operations with them, I get this warning followed by a string of exceptions.

`WARNING:root:Worker me couldn't recognize worker 0`

Has anyone had a similar problem? @iamtrask @mari-linhares

File: grid/websocket_client.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        addr: str,
+        id: Union[int, str] = 0,
+        is_client_worker: bool = False,
+        log_msgs: bool = False,
+        verbose: bool = False,
+        data: List[Union[torch.Tensor, AbstractTensor]] = None,
+    ):
+        """
+        Args:
+            hook (sy.TorchHook): a normal TorchHook object
```
Comment (mari-linhares) [inline grid/websocket_client.py]: tip: if you're using typing, I don't think types are needed at docstrings.

**Comment by mari-linhares** [comment]
Can you send an example of how you're doing it? I would say that probably these tensors are not owned by the same owner. When you send/retrieve an object you'll need to update the tensor's location and owner (for grid workers I don't think this is done automatically).

**Comment by IonesioJunior** [comment]
Problem solved. Just needed to specify the worker id.
[Experimental Notebook](https://github.com/IonesioJunior/Grid/blob/add_websocket_approach/examples/experimental/Launch_Websocket_Grid_worker.ipynb)

File: grid/websocket_client.py (hunk 1 above)
Comment (IonesioJunior) [reply grid/websocket_client.py]: Fixed at ebc7c81

**Pull request merged**